src/main/java/no/nav/veilarboppfolging/service/AuthService.java

```diff
@@ -406,6 +336,106 @@ public static boolean isTokenX(Optional<JWTClaimsSet> maybeClaims) {
+                    } catch (ParseException e) {
+                        return Optional.empty();
+                    }
+                })
+                .orElse(emptyList())
+                .contains("access_as_application");
+    }
+
+    private void sjekkTilgang(ActionId actionId, AktorId aktorId) {
+        Optional<String> sikkerhetsnivaa = hentSikkerhetsnivaa();
+        if (erInternBruker()) {
+            Decision decision = poaoTilgangClient.evaluatePolicy(new NavAnsattTilgangTilEksternBrukerPolicyInput(
+                    hentInnloggetVeilederUUID(), mapActionTypeToTilgangsType(actionId), getFnrOrThrow(aktorId).get()
+            )).getOrThrow();
+            auditLogWithMessageAndDestinationUserId(
+                    "Veileder har gjort oppslag på aktorid",
+                    aktorId.get(),
+                    authContextHolder.getNavIdent().orElse(NavIdent.of(UKJENT_NAV_IDENT)).get(),
+                    decision.isPermit() ? AuthorizationDecision.PERMIT : AuthorizationDecision.DENY
+            );
+            if (decision.isDeny()) {
+                throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+            }
+        } else if (erEksternBruker() && sikkerhetsnivaa.isPresent() && sikkerhetsnivaa.get().equals("Level4")) {
+            Decision decision = poaoTilgangClient.evaluatePolicy(new EksternBrukerTilgangTilEksternBrukerPolicyInput(
+                    hentInnloggetPersonIdent(), getFnrOrThrow(aktorId).get()
+            )).getOrThrow();
+            auditLogWithMessageAndDestinationUserId(
+                    "Ekstern bruker har gjort oppslag på aktorid",
+                    aktorId.get(),
+                    hentInnloggetPersonIdent(),
+                    decision.isPermit() ? AuthorizationDecision.PERMIT : AuthorizationDecision.DENY
+            );
+            if (decision.isDeny()) {
+                throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+            }
+
+        } else if (erSystemBruker()) {
+            if (!veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), actionId, aktorId)) {
+                // TODO: skriv oss bort fra abac for systembruker
+                throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+            }
+        } else {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+        }
+    }
+
+    private boolean erAadOboToken() {
+        Optional<String> navIdentClaim = authContextHolder.getIdTokenClaims()
+                .flatMap((claims) -> authContextHolder.getStringClaim(claims, "NAVident"));
+        return authContextHolder.getIdTokenClaims().map(JWTClaimsSet::getIssuer).filter(environmentProperties.getNaisAadIssuer()::equals).isPresent()
+                && authContextHolder.getIdTokenClaims().map(x -> x.getClaim("oid")).isPresent()
+                && navIdentClaim.isPresent();
+    }
+
+    @SneakyThrows
+    private Optional<NavIdent> getNavIdentClaimHvisTilgjengelig() {
```

Comment:
This function is not in use

Reply:
Dette er eksisterende kode som jeg bare har endret rekkefølgen i filen på.